dvc/stage/__init__.py

```diff
@@ -493,7 +493,11 @@ def _status_stage(self, ret):
             ret.append("changed checksum")
 
     def already_cached(self):
-        return self.deps_cached() and self.outs_cached()
+        return (
+            not self.changed_stage()
+            and self.deps_cached()
+            and self.outs_cached()
+        )
```

Comment:
This does not fixes/affects anything. The user of this API already checked these themselves. I am just moving it where it belongs.